### PR TITLE
Clear space and don't use cxx-common docker image

### PR DESCRIPTION
**Pull request by ekilmer**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,8 +13,6 @@ on:
       - "*"
 
   pull_request:
-    branches:
-      - "*"
```
Comment (Ninja3047) [inline .github/workflows/ci.yml:17]: i think we want this so we have CI run when making PR's to other branches?

File: cmake/utils.cmake [hunk 2]
```diff
@@ -161,7 +161,7 @@ function(InstallExternalTarget target_name target_path install_type installed_fi
 
   install(FILES "${output_file_path}"
     TYPE ${install_type}
-    PERMISSIONS OWNER_READ OWNER_EXECUTE
+    PERMISSIONS OWNER_READ OWNER_EXECUTE OWNER_WRITE
```
Comment (Ninja3047) [inline cmake/utils.cmake:164]: why do we want this?

**Review by 2over12 — APPROVED** [review]
lgtm

File: cmake/utils.cmake (hunk 2 above)
Comment (ekilmer) [reply cmake/utils.cmake:164]: There was an issue with packaging where `strip` could not rewrite the installed file which failed here https://github.com/lifting-bits/remill/actions/runs/6498034215/job/17648485113#step:6:6691
and was resolved by this change here https://github.com/lifting-bits/remill/actions/runs/6498574821/job/17650196369#step:6:6651

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ekilmer) [reply .github/workflows/ci.yml:17]: I had to remove this to get CI to run on my PR

**Pull request merged**